src/lib.rs

```diff
@@ -98,12 +98,21 @@ fn internal_generate(args: GenerateArgs) -> Result<PathBuf> {
     )?;
 
     // the `--init` parameter may also be set by the template itself
-    // TODO: should we show a warning to the user if the template doesn't agree with the parameter? The user might not expect the template files to end up in CWD!
-    user_parsed_input.init |= config
+    if config
         .template
         .as_ref()
         .and_then(|c| c.init)
-        .unwrap_or(false);
+        .unwrap_or(false)
+        && !user_parsed_input.init
+    {
+        eprintln!(
```

Comment:
would it not make sense to prompt the user with a question here, so that he can actively consent to the `--init` behavior or reject it and stop hereby the template processing?

Reply:
I for one would be kinda upset if `cargo-generate` did that to me - the reason for specifying the init behavior in the template is because the template is only designed to use with that behavior, and the user is too lazy to specify it on the command line every time (or too forgetful in my case!).

We've had the TODO in the code for quite a while, to just warn the user that this is happening - so the only behavior changed is the actual warning.

Reply:
also very reasonable, fine with me.